### PR TITLE
fix expression indentation error

### PR DESCRIPTION
**Pull request by TimWhiting**
I've run into this syntax error a lot since I'm not super familiar with elm or roc style programming or functional programming in general.

Replaces #2129

### CODE REVIEW

**Review by JanCVanB — DISMISSED** [review]
I'm not familiar with the parser and I don't understand the macros & big picture involved in your changes, but I'm glad to see the tests pass, including the new one. That's enough for me to approve it, but perhaps we should get a quick look from a parser pro.

@ayazhafiz @bhansconnect @folkertdev are any/all of you parser experts/maintainers? In the future I could tag only one of you, if I knew who's THE parser pro. :)

File: crates/reporting/tests/test_reporting.rs [hunk 1]
```diff
@@ -4586,6 +4586,28 @@ mod test_reporting {
...
+
+    I am partway through parsing an expression, but I got stuck here:
+
+    1│  app "test" provides [main] to "./platform"
+    2│
+    3│  main =
+    4│      f <- Foo.foo
+                        ^
+
+    Looks like the indentation ends prematurely after an expression
```
Comment (ayazhafiz) [inline crates/reporting/tests/test_reporting.rs:4607]: Can we add a period and a suggestion for what we'd expect here, e.g. something like 
```suggestion
    Looks like the indentation ends prematurely here. Did you mean
    to have another expression after this line?
```
?

**Review by ayazhafiz — APPROVED** [review]
One suggestion on a better error message, not blocking; feel free to merge as is.

**Review by JanCVanB — APPROVED** [review]
Hooray! Thank you very much for revisiting this, @TimWhiting 😃

**Comment by Anton-4** [comment]
@TimWhiting can you update commit 805ce04 so it is signed?

**Comment by TimWhiting** [comment]
> @TimWhiting can you update commit [805ce04](https://github.com/roc-lang/roc/commit/805ce046ac14555b704e01e49d8e823123feb493) so it is signed?

Done, I had been working on my desktop before, but made the error message change on my Macbook, then I had some random git problems.

**Comment by JanCVanB** [comment]
@TimWhiting basically the same thing just happened to me on #4064 😁

**Review by Anton-4 — APPROVED** [review]
Thanks @TimWhiting!

**Pull request merged**